### PR TITLE
Fix #1631

### LINKED ISSUE #1631: Splash/Blood on Hole
If any type of splash/blood is on hole you cant open it.
On picture we clearly see that blood is above hole.

### PR DESCRIPTION
**Pull request by ninjalulz**
Players should now be able open a hole even though there are items/splash on top of it.

### CODE REVIEW

File: data/actions/scripts/tools/shovel.lua [hunk 1]
```diff
@@ -1,9 +1,27 @@
...
+	end
+
+	local groundId = ground:getId()
+	if isInArray(holes, groundId) then
+		ground:transform(groundId + 1)
+		ground:decay()
+
+		toPosition.z = toPosition.z + 1
+		tile:relocateTo(toPosition)
+	elseif groundId == 231 then
```
Comment (ranisalt) [inline data/actions/scripts/tools/shovel.lua]: What is 231?

**Comment by ranisalt** [comment]
> and for who use the tools actions at the lib ?

IIRC ninja is the maintainer of ORTS, he probably will update that too as soon as this PR is merged

**Comment by marksamman** [comment]
Are all hole items ground tiles? If not, then this breaks them, but if they are, then this looks good to me.

File: data/actions/scripts/tools/shovel.lua (hunk 1 above)
Comment (ninjalulz) [inline data/actions/scripts/tools/shovel.lua]: It's sand, probably an old id.

**Comment by ninjalulz** [comment]
@marksamman The item ids whom are stored in the holes array are, but there is an id that's not being used that isn't a ground tile.

I could however, update the pull request to use this: https://gist.github.com/ninjalulz/1cc7a6f3106b7ef683bb

**Comment by marksamman** [comment]
That's good enough for now then, getItemById should be avoided if possible since it iterates over all items on the tile.

**Pull request merged**